src/kokkos/plugin-PixelTriplets/kokkos/CAHitNtupletGeneratorKernels.cc

```diff
@@ -45,10 +45,10 @@ namespace KOKKOS_NAMESPACE {
     assert(teamSize > 0 && 0 == teamSize % 16);
     teamSize *= stride;
 
-#ifdef KOKKOS_BACKEND_SERIAL
-    Kokkos::TeamPolicy<KokkosExecSpace> policy{execSpace, leagueSize, 1};
-    // unit stride loop for serial execution
-    stride = 1;
+#if defined KOKKOS_BACKEND_SERIAL || defined KOKKOS_BACKEND_PTHREAD
+    // for host backends use the same stride as the number of threads in the team
+    stride = KokkosExecSpace::impl_thread_pool_size();
+    Kokkos::TeamPolicy<KokkosExecSpace> policy{execSpace, leagueSize, stride};
```

Comment:
@PointKernel Could you double-check my changes in this file?

Reply:
Does this give the correct numerical results? I can understand if the pthread policy is defined in a CUDA-similar way, i.e. adjusting team size and league size based on a predefined stride size.  Making the stride size equal to team size will theoretically work cause the kernels involved (like `fishbone`) are just to iterate to a certain upper bound. But the algorithm is different from the original CUDA implementation. It's hard to tell without evaluating the eventual output.

Reply:
I'm somewhat convinced that the results are correct. At least the pthread backend passes the counting-based validation (which has been rather picky) at least up to 8 threads. I think the `stride` mostly adjusts the balance between the two nested loops in the `getDoublets`, `kernel_connect`, and `fishbone` kernels.

Reply:
I could actually just leave the stride 1 for pthread backend as well, on a quick test it in fact appears to scale better